### PR TITLE
Deprecate `Javadoc.isVerbose`/`setVerbose`

### LINKED ISSUE #30308: Deprecate `Javadoc.isVerbose/setVerbose` and prefer verbose setting on `CoreJavadocOptions`
These properties should belong to options. Also due to werid setters is hard to migrate them to provider API.

https://github.com/gradle/gradle/blob/678cf709a77f9f432768c6c0ee7eb48a0ddaed91/platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java#L311-L327

### PR DESCRIPTION
**Pull request by octylFractal**
Fixes #30308

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Review by lkasso — APPROVED** [review]

File: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc [hunk 1]
```diff
@@ -142,6 +142,11 @@ The existing method does not behave as specified by its javadoc, and we do not p
 
 Potential migrations include using `Object.equals(Object)` directly, or comparing the fields of dependencies manually.
 
+[[deprecated_javadoc_verbose]]
+==== Deprecated `Javadoc.isVerbose()` and `Javadoc.setVerbose(boolean)`
+
+These methods duplicate the functionality of `Javadoc.getOptions().isVerbose()` and `Javadoc.getOptions().verbose()`, and do not document that setting `false` does not do anything.
```
Comment (big-guy) [inline platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc:148]: Can we change this a bit to spell out the migration?

Something like:
```
These methods have been deprecated on link:<link to javadoc class>[`Javadoc`]:
- link:<link to method in javadoc>[`isVerbose()`] is replaced by link:<link to JavadocOptions.isVerbose>
- link:<link to method in javadoc>[`setVerbose(boolean)`] is replaced by link:<link to JavadocOptions.verbose>
```

File: platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java [hunk 2]
```diff
@@ -309,8 +311,13 @@ public void setTitle(@Nullable String title) {
...
      */
     @Internal
-    @ToBeReplacedByLazyProperty
+    @NotToBeReplacedByLazyProperty(because = "deprecated and replaced by options.verbose")
     public boolean isVerbose() {
+        DeprecationLogger.deprecateMethod(Javadoc.class, "isVerbose()")
+            .withAdvice("Please use the getOptions().isVerbose() method instead. It may be necessary to cast the options to CoreJavadocOptions to access the replacement method.")
+            .willBeRemovedInGradle9()
+            .withUpgradeGuideSection(8, "deprecated_javadoc_verbose")
+            .nagUser();
```
Comment (big-guy) [inline platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java:320]:
```suggestion
        DeprecationLogger.deprecateMethod(Javadoc.class, "isVerbose()")
            .replaceWith("getOptions().isVerbose()")
            .willBeRemovedInGradle9()
            .withUpgradeGuideSection(8, "deprecated_javadoc_verbose")
            .nagUser();
```

File: platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java [hunk 3]
```diff
@@ -322,7 +329,18 @@ public boolean isVerbose() {
      */
     public void setVerbose(boolean verbose) {
         if (verbose) {
+            DeprecationLogger.deprecateMethod(Javadoc.class, "setVerbose(true)")
+                .withAdvice("Please use the getOptions().verbose() method instead. It may be necessary to cast the options to CoreJavadocOptions to access the replacement method.")
```
Comment (big-guy) [inline platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java:333]: Why would casting this be necessary? It looks like `MinimalJavadocOptions` has the method?

**Review by big-guy — CHANGES_REQUESTED** [review]

File: platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java (hunk 3 above)
Comment (octylFractal) [reply platforms/jvm/language-java/src/main/java/org/gradle/api/tasks/javadoc/Javadoc.java:333]: Ah, you're right. I'll remove this.

**Review by big-guy — APPROVED** [review]

**Pull request merged**